### DISCUSSION TITLE
Scan on Weekly Data

### DISCUSSION THREAD

**Discussion by cotton1234** [BennyThadikaran/stock-pattern#49] [General]
Hello Benny,

If i want to scan the same on weekly data , anything to change or I just recalculate the OHLC on weekly time frame.

Thanks
Saurabh

**Comment by BennyThadikaran** [comment]
Hi Saurabh,

Just run `py init.py --tf weekly`

Assume you have CSV files stored in daily timeframe, you can use the `--tf` option to specify `weekly` or `monthly`. The default is `daily`.

**Reply by cotton1234** [reply]
Thanks Benny. Works. Will check the charts.

**Comment by Bhanujha** [comment]
I have tried using --tf 1 , --tf 15; the only thing that changes is headline where it changes from default 15 to 1 or 4h but the actual time frame remains daily. Also how to get intraday daya. I download the OHLC data from EOD as mentioned in your description

**Reply by Bhanujha** [reply]
Also I have changed my default loader to IEODFileLoader and set default timer to 15 min but still the output is daily timeframe when running py init.py --tf 1

**Comment by BennyThadikaran** [comment]
Hi Bhanujha,

If you have daily timeframe CSV data its not possible to switch to a lower timeframe. You can only switch upwards i.e weekly or monthly.

If you have 1 min CSV data you can switch to any timeframe above it. 

If you had 15 min CSV data, you can switch to 60, 120 240 or even daily provided you have enough data. But timeframes lower than 15 wont be possible. So 1min, 5min is out.

The DEFAULT_TF is referring to the timeframe of your CSV data. Hope that helps.

The docs may not be very clear on this. I'll see what can be done.

**Comment by cotton1234** [comment]
NSE website will not give any intraday data.
We need to use Broker api to download the same.
I use Fyers to get intraday day to run my automations.

All brokers now offer the same.

Thanks
Saurabh

**Reply by Bhanujha** [reply]
How did you download from fyers?

**Reply by cotton1234** [reply]
https://myapi.fyers.in/docsv3 Pls follow the same.

Thanks

**Reply by Bhanujha** [reply]
I have tried to extract 1 min OHLC data from Dhan API but don't understand how to use it to view in plotter or init. @BennyThadikaran 
I tried to copy the data as CSV and then separate it into columns using delimiters. How am I supposed to update the data automatically daily and let init.py use it to show charts?

Also as the SYS list is present for NSE-500 data and the Data path is saved to eod_data/daily with a daily timeframe do I have to change the data path from daily to create a new folder intraday and add it as a data path? (Tried this but always get an error the file was not found). I only tried with 1 in stock. rest it will show the file was not found but it also showed the same for pnb. I ensured the CSV file name was in small letters.

**Reply by Bhanujha** [reply]
@cotton1234 I can extract single stock 1 min timeframe data at a time is there a way to get all NSE-500 stocks 1 min timeframe data at a time? Where I can keep it in a loop where new day data is added and the last day data keeps on getting deleted

**Reply by cotton1234** [reply]
> @cotton1234 I can extract single stock 1 min timeframe data at a time is there a way to get all NSE-500 stocks 1 min timeframe data at a time? Where I can keep it in a loop where new day data is added and the last day data keeps on getting deleted

You cant get 1 min data in 1 go. You need to do loop only. Unless you subscribe to some 3rd party to get the csv files of the same.
See if you can use file based DB(SQLite) and keep storing the data instead of deletion. 

Thanks
Saurabh

**Reply by Bhanujha** [reply]
'You can get 1 min data in 1 go' for a single stock only or for NSE_500 all stocks at a time? @cotton1234

**Reply by Bhanujha** [reply]
What you have done to use for data analysis? @cotton1234.

**Reply by cotton1234** [reply]
> What you have done to use for data analysis? @cotton1234.

I used it for some other purpose. not intraday data.  Just gave you idea.
